opensquirrel/squirrel_ir.py

```diff
@@ -148,6 +148,14 @@ def name(self) -> Optional[str]:
 
     @property
     def is_anonymous(self) -> bool:
```

Comment:
Maybe add a doc for this function? It's quite readable, but I think it would help. 

So, to check if a gate is anonymous, we walk all the default gate set, filter those which have one argument, assume that that one argument is a qubit (maybe worth adding a check/assert for this?), check that the axis and phase of that qubit argument for the default gate is the same as ours, and, in that case, convert ourselves to that default gate?

But there's something I don't like about this code. It has lateral effects. The functions says `is_anonymous`, but it should be called something like `convert_me_to_a_default_gate_if_possible`. I suggest this is done through different functions. E.g.:
- `is_anonymous` just checks if the gate is anonymous or not. I don't know how useful this function can be.
- `is_close_to_a_default_gate` could return `None` if it's  not, or `Gate` if it is (with the code below).
- `try_name_anonymous_gate(s)` could be a free function that goes through one/many anonymous gates and, if they are close to a default gate, name them (convert  them).

Reply:
I have now added a _check_if_anonymous function that does the check. I do believe it should return true or false. This function is a bit strange because it accepts a `BlochSphereRotation`, not a `Gate`.